eclair-core/src/main/scala/fr/acinq/eclair/wire/LightningMessageCodecs.scala

```diff
@@ -65,6 +66,13 @@ object LightningMessageCodecs {
   // we have limited space for backup, largest message is commit_sig with 30 htlcs in each direction: 65535B - (32B + 64B + 2*30*64B) = 61599B ~= 60000B
   val channeldataoptional: Codec[Option[ByteVector]] = choice(optional(magic, limitedSizeBytes(60000, variableSizeBytesLong(varintoverflow, bytes))), provide(Option.empty[ByteVector]))
 
+  // TODO: same hack as above, we use a carefully chosen magic to ensure compatibility with TLV encoding
```

Comment:
I'm really going to enjoy it when we delete that branch :sweat_smile:

Reply:
FWIW, we have TLV support on this branch, but going with magic was so much easier 🤷‍♂️